tokenizers/javascript/index.js

```diff
@@ -0,0 +1,88 @@
+const esprima = require('esprima')
+const escodegen = require('escodegen')
+const fs = require('fs-extra-promise')
+const tokenizer = require('./tokenizer')
+
+const immutable = require('immutable')
+const walk = require('esprima-walk')
+
+const { base64FileName } = require('./util')
+
+const estools = require('estools')
+
+const TOKENIZER_SCOPE_FILE = 'file-scope'
+const TOKENIZER_SCOPE_FUNCTION = 'function-scope'
+
+const TOKENIZER_SCOPE = TOKENIZER_SCOPE_FILE
+
+// TODO: estools map / filter / traverse (instead of walk)
+// - filter subfunctions from fuction asts somehow
+// - test on SCC
+
+// TODO: get rid of the function block and indentation
+const regenerateFunctionCode = function(functionAst) {
+  codegenOptions = { // NOTE: doesn't help
+    format: {
+      parentheses: false
+    }
+  }
+
+  // NOTE: functionAst.body ommits the function signature (returns block only)
+  return escodegen.generate(functionAst.body, {})
```

Comment:
It is possible. Is is needed though? Do other tokenizers include the signature? Maybe I'll add it into config file.

Because when we do queries with SourcererCC, e.g. snippets from StackOverflow, I think the signature is not always good idea for finding a match, especially when the snippet is only few lines of simple expressions.